### PR TITLE
polybar: 3.5.2 -> 3.5.5

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/polybar/default.nix [hunk 1]
```diff
@@ -23,7 +24,7 @@
 # disable modules
 , alsaSupport   ? true,  alsaLib       ? null
 , githubSupport ? false, curl          ? null
-, mpdSupport    ? false, libmpdclient ? null
+, mpdSupport    ? false, libmpdclient  ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/polybar/default.nix:27]:
```suggestion
, mpdSupport    ? false, libmpdclient
```

Together with the assert is not required within the scope of nixpkgs.

Comment (fortuneteller2k) [reply pkgs/applications/misc/polybar/default.nix]: I'll have it in a separate commit.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120021` at 158f03a6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>polybar</li>
    <li>polybarFull</li>
  </ul>
</details>

**Review by Br1ght0ne — APPROVED** [review]
Diff LGTM, packages built and working.

**Review by fufexan — APPROVED** [review]

File: pkgs/applications/misc/polybar/default.nix [hunk 2]
```diff
@@ -18,41 +19,48 @@
...
       fetchSubmodules = true;
     };
 
+    nativeBuildInputs = [
+      cmake
+      pkg-config
+      python3Packages.sphinx
+      removeReferencesTo
+
+      (if i3Support || i3GapsSupport then makeWrapper else null)
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/polybar/default.nix:61]: Why not use lib.optional here instead?

File: pkgs/applications/misc/polybar/default.nix [hunk 3]
```diff
@@ -79,8 +87,6 @@ stdenv.mkDerivation rec {
       (if i3Support || i3GapsSupport then jsoncpp else null)
       (if i3Support then i3 else null)
       (if i3GapsSupport then i3-gaps else null)
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/polybar/default.nix:89]: Please replace this if then null with lib.optional(s).

**Pull request merged**